src/mjolnir/landmark_database_builder.cc

```diff
@@ -1,103 +1,114 @@
 #include "mjolnir/landmark_database_builder.h"
+#include "filesystem.h"
 #include "mjolnir/util.h"
 
 namespace valhalla {
 namespace mjolnir {
-void LandmarkDatabase::connect_database() {
-  if (!open_database()) {
-    throw std::runtime_error("Cannot open database");
-    return;
-  }
 
-  if (open_flags == (SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE)) {
-    if (!create_landmarks_table() || !create_spatial_index()) {
-      throw std::runtime_error("Cannot create landmarks table or spatial index");
-      return;
+// TODO: this can be a utility and be more generic with a few more options, we could make the prepared
+//  statements on the fly and retrievable by the caller, then anything in the code base that wants to
+//  use sqlite can make use of this utility class. for now its ok to be specific to landmarks though
+struct LandmarkDatabase::db_pimpl {
```

Comment:
i think this class could get more generic and let users of it register prepared statements, retrieve them and run them. then we could use this class for any sqlite needs. for now i didnt go that far but if we do then this can go to a private header